lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -375,7 +373,8 @@ static aievec::CmpOp createCmpOpAieML(ConversionPatternRewriter &rewriter,
 template <typename DstOpTy>
 static void generateAIEVecOpsForReductionOp(ConversionPatternRewriter &rewriter,
                                             vector::ReductionOp srcOp,
-                                            int shiftIndex, Value curValue) {
+                                            unsigned shiftIndex,
+                                            Value curValue) {
```

Comment:
Narrowing from int to unsigned is implementation dependent.

Reply:
Why are we switching to "unsigned" for `shiftIndex` at all? I'm not a fan of unsigned types unless they're strictly necessary, signed types are generally safer to use.

Reply:
The root cause is some of the upstream APIs return unsigned (e.g., `getPosition`) and so that carries forward and leads to narrowing conversions. I guess you're saying signed is better because it overflows rather than ... (not sure what happens for unsigned - is it UB?).